core/src/banking_stage/latest_unprocessed_votes.rs

```diff
@@ -343,8 +362,8 @@ impl LatestUnprocessedVotes {
         if bank.epoch() <= self.current_epoch.load(Ordering::Relaxed) {
             return;
         }
-        let mut staked_nodes = self.cached_staked_nodes.write().unwrap();
-        *staked_nodes = bank.current_epoch_staked_nodes().clone();
+        let mut epoch_stakes = self.cached_epoch_stakes.write().unwrap();
+        *epoch_stakes = bank.current_epoch_stakes().clone();
```

Comment:
should we hold this exclusive lock for the below eviction period?
can/should we drop and re-acquire the read-lock?

Reply:
The only other competitor is `insert_batch` on the gossip thread, since `cache_epoch_boundary_info` must be run by the tpu thread. 

I don't think it'll make too much of a difference, but probably best to limit our lock holding anyway. It's not necessary from a correctness perspective to hold both write locks at the same time.